vcd/resource_vcd_lb_server_pool.go

```diff
@@ -0,0 +1,346 @@
+package vcd
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/vmware/go-vcloud-director/v2/types/v56"
+)
+
+func resourceVcdLBServerPool() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceVcdLBServerPoolCreate,
+		Read:   resourceVcdLBServerPoolRead,
+		Update: resourceVcdLBServerPoolUpdate,
+		Delete: resourceVcdLBServerPoolDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceVcdLBServerPoolImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"vdc": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD organization in which the Service Monitor is located",
+			},
+			"org": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "vCD virtual datacenter in which the Service Monitor is located",
+			},
+			"edge_gateway": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Edge gateway name in which the Server Pool is located",
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Unique Server Pool name",
+			},
+
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Server pool description",
+			},
+			"algorithm": &schema.Schema{
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Balancing method for the service. One of ip-hash, round-robin, uri, leastconn, url, or httpheader",
+				ValidateFunc: validateCase("lower"),
+			},
+			"algorithm_parameters": {
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Additional options for load balancing algorithm for http-header or url algorithms",
+			},
+			"monitor_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Load Balancer Service Monitor ID",
+			},
+			"enable_transparency": &schema.Schema{
+				Type:        schema.TypeBool,
+				Default:     false,
+				Optional:    true,
+				Description: "Makes client IP addresses visible to the backend servers",
+			},
+
+			"member": {
+				Optional: true,
+				ForceNew: false,
+				Type:     schema.TypeList,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							ForceNew:    false,
+							Computed:    true,
+							Type:        schema.TypeString,
+							Description: "Pool member id",
+						},
+						"condition": &schema.Schema{
+							Type:         schema.TypeString,
+							ForceNew:     false,
+							Required:     true,
+							ValidateFunc: validateCase("lower"),
+							Description:  "Defines member state. One of enabled, drain, disabled.",
```

Comment:
No validation on a list of values?

Reply:
Can add but again I relied on API errors:

```
vcd_lb_server_pool.web-servers: Modifying... [id=pool-94]

Error: unable to update load balancer server pool with ID pool-94: error while updating load balancer server pool : vShield Edge [LoadBalancer] Invalid member condition asd. Valid member conditions are: enabled|drain|disabled. (API error: 14555)
```